### ISSUE TITLE
GTK failures with overwitch 1.2

### ISSUE THREAD

**Issue by ScottGarman** [dagargo/overwitch#68]
Hi there, on my Manjaro system I have found that overwitch 1.2 does not run and shows the following gtk failure messages when I try to run it:

```
(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_about_dialog_set_version: assertion 'GTK_IS_ABOUT_DIALOG (about)' failed

(overwitch:650733): GLib-GObject-CRITICAL **: 20:32:29.612: invalid (NULL) pointer instance

(overwitch:650733): GLib-GObject-CRITICAL **: 20:32:29.612: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(overwitch:650733): GLib-GObject-CRITICAL **: 20:32:29.612: invalid (NULL) pointer instance

(overwitch:650733): GLib-GObject-CRITICAL **: 20:32:29.612: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(overwitch:650733): GLib-GObject-CRITICAL **: 20:32:29.612: invalid (NULL) pointer instance

(overwitch:650733): GLib-GObject-CRITICAL **: 20:32:29.612: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(overwitch:650733): GLib-GObject-CRITICAL **: 20:32:29.612: invalid (NULL) pointer instance

(overwitch:650733): GLib-GObject-CRITICAL **: 20:32:29.612: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(overwitch:650733): GLib-GObject-CRITICAL **: 20:32:29.612: invalid (NULL) pointer instance

(overwitch:650733): GLib-GObject-CRITICAL **: 20:32:29.612: g_signal_connect_data: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_application_add_window: assertion 'GTK_IS_WINDOW (window)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_column_view_column_set_visible: assertion 'GTK_IS_COLUMN_VIEW_COLUMN (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_column_view_column_set_visible: assertion 'GTK_IS_COLUMN_VIEW_COLUMN (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_column_view_column_set_visible: assertion 'GTK_IS_COLUMN_VIEW_COLUMN (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_column_view_column_set_visible: assertion 'GTK_IS_COLUMN_VIEW_COLUMN (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_column_view_column_set_visible: assertion 'GTK_IS_COLUMN_VIEW_COLUMN (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_spin_button_set_value: assertion 'GTK_IS_SPIN_BUTTON (spin_button)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_spin_button_set_value: assertion 'GTK_IS_SPIN_BUTTON (spin_button)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_drop_down_set_selected: assertion 'GTK_IS_DROP_DOWN (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_column_view_column_set_visible: assertion 'GTK_IS_COLUMN_VIEW_COLUMN (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_column_view_column_set_visible: assertion 'GTK_IS_COLUMN_VIEW_COLUMN (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_column_view_column_set_visible: assertion 'GTK_IS_COLUMN_VIEW_COLUMN (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_column_view_column_set_visible: assertion 'GTK_IS_COLUMN_VIEW_COLUMN (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.612: gtk_column_view_column_set_visible: assertion 'GTK_IS_COLUMN_VIEW_COLUMN (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.617: gtk_window_present: assertion 'GTK_IS_WINDOW (window)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.620: gtk_label_set_text: assertion 'GTK_IS_LABEL (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.620: gtk_label_set_text: assertion 'GTK_IS_LABEL (self)' failed

(overwitch:650733): Gtk-CRITICAL **: 20:32:29.620: gtk_label_set_text: assertion 'GTK_IS_LABEL (self)' failed
```

I am able to use overwitch-cli without problems.

If I build overwitch 1.1, these errors do not occur and I am able to run the GUI app without problems.

My system is using the gtk4 1:4.14.5-1 package from Manjaro stable. I will attach my config.log file, let me know if I can provide any other relevant info. And thank you for your work on overbridge!

**Comment by ScottGarman** [comment]
[config.log](https://github.com/user-attachments/files/17613807/config.log)

**Comment by dagargo** [comment]
Current `master` branch version is 2.0. (I committed this after you open the issue. Version 1.2 is obsolete as of now.)

> If I build overwitch 1.1, these errors do not occur and I am able to run the GUI app without problems.

That's really weird and the errors above don't make any sense. Everything seems to be right regarding GTK4.

Could it be that you forgot to run `./configure` with no parameters and somehow the ui file being red is from another version? This does not make sense either because `overwitch` is installed along with the right GUI file too but...

Can you please switch to `master`, try the whole official installation instructions and report back?

```
autoreconf --install
./configure
make
sudo make install
sudo ldconfig
```

**Comment by dagargo** [comment]
I've been able to replicate the issue on a fresh Fedora 40 installation using `master`. I'll continue tomorrow.

**Comment by dagargo** [comment]
The issue is in Overwitch. The problem is that Overwitch requires GTK4 version 4.16 and not 4.12 as it is stated here.

https://github.com/dagargo/overwitch/blob/20441d2608155d6f0dd56f1e5fe3bea3e481d4ff/configure.ac#L20

With this fix, compilation wouldn't even take place in Fedora 40 as `./configure` would fail.

(I've already fixed that in ae35246edc61736745e0593c6ae1ad943864ccb3.)

I've checked that GTK4 version 4.16 is already in Fedora 41 so that upgrading would be a possibility.
But as I'm not familiar with Fedora I'd like to ast if it is 41 too bleeding edge from the users point of view.

I'm sorry for choosing the latest version of GTK4 as it imposes somehow a constraint for you but now I'm can't go back and needed to move on with all the changes in libraries and tools. It's just that I was not expecting the Overbridge changes that have messed with my planning.

I've been thinking on creating a branch based on 1.1 (GTK3) with all the new Overbridge commits merged into it. In fact, I've done it already and it works but it's not pushed yet.

Is publishing this in a branch as a temporary solution a good idea for you? It could also be an official version 1.1.1.

I think I like this later approach because other people are having issues with this too when running on distributions as Debian stable.

What are you thoughts on this?

**Comment by ThaYke** [comment]
I'm totally fine with whatever solution you can offer. :) A branch would be nice.

I think to require the latest Fedora is a bit though sometimes, as 41 just released, and there can be a few issues, especially on a system for audio production which can depend on quite a few different packages from different repos.

This is totally subjective, of course, but I think to support the latest and second latest Fedora would ease the pressure to update a bit, especially regarding those transitions between versions.

**Comment by dagargo** [comment]
Branch `gtk3-overbridge-cc-midi` based on version 1.1 (GTK3) will include all the changes related to the new USB class compliant MIDI that will get into `master` (GTK4 and version 2.0).

The only purpose of that branch is to release version 1.1.1 that will allow users with relatively old libraries to compile and run Overwitch. Very likely, there will be no further development.

Version 1.1.1 will be releases together with version 2.0.

@ScottGarman and @ThaYke, let me know if this fixes your issues.

**Comment by ThaYke** [comment]
I didn't fully test it in action, but said errors are gone in the new branch, and overwitch starts up just fine for me. Thank you for your help!

**Comment by dagargo** [comment]
I'm closing this as completed.

Branch `gtk3-overbridge-cc-midi`, which is based on GTK3, will support all new Overbridge devices firmwares. It will eventually be released as version 1.1.1.

**Comment by ScottGarman** [comment]
Apologies for the late reply, I promptly forgot I had filed this. Thank you for the fix, and I run Manjaro stable, which now has gtk 4.16, so this wouldn't block me in any case. :+1:
